sympy/solvers/solvers.py

```diff
@@ -1758,17 +1758,6 @@ def _solve_system(exprs, symbols, **flags):
                 for e in subexpr:
                     subsyms |= exprsyms[e]
                 subsyms = list(sorted(subsyms, key = lambda x: sym_indices[x]))
-                # use canonical subset to solve these equations
```

Comment:
How is this situation being handled now?

Reply:
I don't understand what situation the deleted code handles.

The code in this block is intended to split a system that can be solved as smaller independent systems e.g. if we have:
```
x + y
x - y - 1
w + t
w - t - 1
```
then it's clear that we can solve the first two equations for x and y separately from the other two equations which can be solved for w and t.

That's the purpose of this block. What I don't understand is the purpose of the code that is deleted in this PR...

Reply:
What happens if you have
```
solve((
x + y,
1/x + 1/y,
x - y - 2))
```

Reply:
For that particular example the result is the same in both PR and master:
```
In [1]: solve((
   ...: x + y,
   ...: 1/x + 1/y,
   ...: x - y - 2))
Out[1]: {x: 1, y: -1}
```

Reply:
if this gives `[]` then I it seems that all is ok:
```
solve((
    x + y,
    x*y,
    x - y - 2))
```

Reply:
Yes, that is the same on master and in the PR:
```python
In [1]: solve((
   ...:     x + y,
   ...:     x*y,
   ...:     x - y - 2))
Out[1]: []
```

Reply:
@smichr are you happy with this change then?

The PR is not ready to be merged because I'll add some more tests both from #21882 and the examples you've just listed here.

Reply:
> are you happy with this change then

I don't see any issues. I like the idea of avoiding solutions that require division by expressions with free symbols. I would like to see the direction (eventually) that solves #17186. Splitting the linear from the non-linear before diving into the non-linear part is important.

Reply:
Yes, I want to have a think about that before merging this change. I think that something like strongly connected components could be used somehow but I haven't quite figured out the right way to pose the system as a directed graph.

Reply:
> something like strongly connected components

There is potentially a related idea [here](https://github.com/sympy/sympy/issues/6662#issuecomment-895382853) in what Tim Peters was directing me towards as a means of doing a derangement efficiently.